content/en/blog/_posts/2021-04-16-volume-health-monitoring-alpha.md

```diff
@@ -0,0 +1,87 @@
+
+## What is Volume Health Monitoring?
+
+CSI Volume Health Monitoring allows CSI Drivers to detect abnormal volume conditions from the underlying storage systems and report them as events on PVCs or Pods.
+
+The Kubernetes components that monitor the volumes and report events with volume health information include the following:
+
+* In addition to gathering existing volume stats, Kubelet also watches volume health of the PVCs on that node. If a PVC has an abnormal health condition, an event will to reported on the pod object that is using the PVC. If multiple pods are using the same PVC, events will be reported on multiple pods.
+* An [external volume health monitor controller](https://github.com/kubernetes-csi/external-health-monitor) watches volume health of the PVCs and reports events on the PVCs.
+
+Note that the node side volume health monitoring logic was an external agent when this feature was first introduced in Kubernetes 1.19 release. In Kubernetes 1.21 release, the node side volume health monitoring logic was moved from the external agent to Kubelet to avoid duplicate CSI function calls. With this change in 1.21, a new alpha feature gate `CSIVolumeHealth` was introduced for the volume health monitoring logic in Kubelet.
+
+Currently the Volume Health Monitoring feature is informational only as it  only reports abnormal volume health events on PVCs or Pods. Users will need to check these events and manually fix the problems. This feature serves as a stepping stone towards programmatic detection and resolution of volume health issues by Kubernetes in the future.
+
+## How do I use Volume Health on my Kubernetes Cluster?
+
+To learn how to use External Volume Health Monitoring Controller, see details in this [doc](
+https://kubernetes-csi.github.io/docs/external-health-monitor-controller.html).
+
+To enable Volume Health Monitoring from the node side, this alpha feature gate `CSIVolumeHealth` needs to be enabled.
+
+If a CSI driver has implemented the Volume Health Monitoring feature from the controller side, events regarding abnormal volume conditions will be recorded on PVCs.
+
+External Health Monitor Controller will also watch for node failure events. This can be enabled by setting the `enable-node-watcher` flag to true. When a node failure event is detected, an event will be reported on the PVC to indicate that pods using this PVC are on a failed node.
+
+If a CSI driver has implemented the Volume Health Monitoring feature from the node side, events regarding abnormal volume conditions will be recorded on pods using the PVCs.
+
+Refer to this [CSI drivers doc](https://kubernetes-csi.github.io/docs/drivers.html) to find out which CSI drivers have implemented this feature.
+
+## As a storage vendor, how do I add support for volume health to my CSI driver?
```

Comment:
It may be good to start off this section by saying that there's a controller-side and node-side. And then split the section into 2 for each side.

Reply:
Split into 2 sections.